drift_dev/lib/src/writer/manager/manager_templates.dart

```diff
@@ -0,0 +1,387 @@
+  ///
+  /// E.g. `AppDatabase get managers => AppDatabaseManager(this);`
+  String databaseManagerGetter(String dbClassName) {
+    return '${databaseManagerName(dbClassName)} get managers => ${databaseManagerName(dbClassName)}(this);';
+  }
+
+  /// The code for a root table manager getter which will be added to the database manager class
+  ///
+  /// E.g. `UserTableManager get todos => UserTableManager(_db,_db.todos);`
+  String rootTableManagerGetter(
+      DriftTable table, String rootTableManagerClass) {
+    return '$rootTableManagerClass get ${table.dbGetterName} => $rootTableManagerClass(_db, _db.${table.dbGetterName});';
+  }
+
+  /// Returns the name of the root manager class for a table
+  ///
+  /// One of these classes is generated for each table in the database
+  ///
+  /// E.g. `\$UserTableManager`
+  String rootTableManagerName(DriftTable table) {
+    return '\$${table.entityInfoName}TableManager';
+  }
+
+  /// Returns the name of the manager class for a table
+  ///
+  /// When using modular generation the manager class will contain the correct prefix
+  /// to access the table manager
+  ///
+  /// E.g. `i0.UserTableTableManager` or `\$UserTableTableManager`
+  String rootTableManagerWithPrefix(DriftTable table, TextEmitter leaf) {
+    return leaf
+        .dartCode(leaf.generatedElement(table, rootTableManagerName(table)));
+  }
+
+  /// Returns the name of the processed table manager class for a table
+  ///
+  /// This does not contain any prefixes, as this will always be generated in the same file
+  /// as the table manager and is not used outside of the file
+  ///
+  /// E.g. `$UserTableProcessedTableManager`
+  String processedTableManagerName(DriftTable table) {
+    return '\$${table.entityInfoName}ProcessedTableManager';
+  }
+
+  /// Class which represents a table in the database
+  /// Contains the prefix if the generation is modular
+  /// E.g. `i0.UserTable`
+  String tableClassWithPrefix(DriftTable table, TextEmitter leaf) =>
+      leaf.dartCode(leaf.entityInfoType(table));
+
+  /// Class which represents a row in the table
+  /// Contains the prefix if the generation is modular
+  /// E.g. `i0.User`
+  String rowClassWithPrefix(DriftTable table, TextEmitter leaf) =>
+      leaf.dartCode(leaf.writer.rowType(table));
+
+  /// Name of this tables filter composer class
+  String filterComposerNameWithPrefix(DriftTable table, TextEmitter leaf) {
+    return leaf
+        .dartCode(leaf.generatedElement(table, filterComposerName(table)));
+  }
+
+  /// Name of this tables filter composer class
+  String filterComposerName(
+    DriftTable table,
+  ) {
+    return '\$${table.entityInfoName}FilterComposer';
+  }
+
+  /// Name of this tables ordering composer class
+  String orderingComposerNameWithPrefix(DriftTable table, TextEmitter leaf) {
+    return leaf
+        .dartCode(leaf.generatedElement(table, orderingComposerName(table)));
+  }
```

Comment:
If possible, can we only use the `withPrefix` variants and remove the text-based versions? That makes things a bit simpler since it's impossible to forget to use `withPrefix` in modular mode. `generatedElement` will not emit a prefix when not necessary so we can use it for the monolithic builder as well.

Reply:
No, even in modular mode you need the name without the prefix when creating the actual class.

```dart
class i0.ClassName { // Invalid
...
}
```

What you're describing happened to me, but any bug like that would throw a static error, which tests find instantaneously